src/lib/store.ts

```diff
@@ -0,0 +1,20 @@
+import type { Piece } from "$lib/models/pieces";
+import { writable, derived } from "svelte/store";
+
+export const selectedPiece = writable<Piece | null>();
+
+export const capturedPiece = writable<Piece>();
+export const capturedBlackPieces = writable<Piece[] | null[]>([]);
```

Comment:
Shouldn't this be Piece[]? I don't foresee this being null

Reply:
isn't it null prior to pieces being there?